### ISSUE TITLE
The context menu show incorrect options in some files

### ISSUE THREAD

**Issue by Saafo** [sxyazi/yazi#700]
### What system are you running Yazi on?

macOS

### What terminal are you running Yazi in?

Warp / iTerm2

### Yazi version

0.2.3

### Did you try the latest main branch to see if the problem has already been fixed?

Not tried, and I'll explain why below

### Describe the bug

> Using main branch need to build yazi locally and need many dependencies. And I've checked recent open and closed issues don't include this.

The options in the open options menu of some files are incorrect, for example:

I've got a `print.sh` and a `print.swift` and their options are incorrect:
<img width="241" alt="image" src="https://github.com/sxyazi/yazi/assets/46736350/1d1c9b11-b92e-4b9c-8297-dc190d260bfc">
<img width="210" alt="image" src="https://github.com/sxyazi/yazi/assets/46736350/e8bedece-204b-446b-bf8c-1c465501e036">

Then I create a `print2.sh` and this time the options are correct:
<img width="190" alt="image" src="https://github.com/sxyazi/yazi/assets/46736350/5a9bb03e-4f08-49cf-a1bc-2dfdbb3af5b2">

Their attributes are basically the same:
<img width="525" alt="image" src="https://github.com/sxyazi/yazi/assets/46736350/75778717-356e-488c-bb4e-a936701f9324">

Besides, I've got a `mov.mp4` file and its options are incorrect either(but another file named `fmp4.mp4` is correct):
<img width="148" alt="image" src="https://github.com/sxyazi/yazi/assets/46736350/4ffd93ef-81e9-4da9-95fc-e391a468cd1e">

And here is the file list in yazi:
<img width="358" alt="image" src="https://github.com/sxyazi/yazi/assets/46736350/83191509-d3d2-41ee-99b4-0e65fd75de7a">
It's interesting that one of the two mp4 files is yellow, and one of the two shell files is yellow. I'm using the default configuration.

### Expected Behavior

All shell files and swift files should contains the `$EDITOR` option, and all video files should contains the play option(mpv)

### To Reproduce

I've got no idea temporarily, do you have and ideas about how this happened?

### Configuration

no config

**Comment by sxyazi** [comment]
Please do `file -bL --mime-type filename` on these files, and then paste the output here.

**Comment by Saafo** [comment]
<img width="290" alt="image" src="https://github.com/sxyazi/yazi/assets/46736350/20b26903-04a4-41bf-9149-ae4e7f54e256">

**Comment by sxyazi** [comment]
That's so weird, could you upload those files and let me try to reproduce it?

**Comment by Saafo** [comment]
Yes! It's so weird. I've tried to copy `print.sh` to another folder, and the options are correct.
[print.sh.zip](https://github.com/sxyazi/yazi/files/14330453/print.sh.zip)

**Comment by evpeople** [comment]
Could this be due to extended attributes in macOS? I noticed that the output of ls -al now includes an @ symbol.

**Comment by evpeople** [comment]
![image](https://github.com/sxyazi/yazi/assets/54983536/4efba88a-fe90-44b2-a033-3e1ff9db8491)
After downloading the print.sh file, I noticed such a symbol. Is this expected behavior? Initially, I thought it was because I used the / for searching, which caused this, but after exiting Yazi and searching for other files, I couldn't replicate this behavior

**Comment by sxyazi** [comment]
> ![image](https://private-user-images.githubusercontent.com/54983536/305936149-4efba88a-fe90-44b2-a033-3e1ff9db8491.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgzNDQ0MzMsIm5iZiI6MTcwODM0NDEzMywicGF0aCI6Ii81NDk4MzUzNi8zMDU5MzYxNDktNGVmYmE4OGEtZmU5MC00NGIyLWEwMzMtM2UxZmY5ZGI4NDkxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE5VDEyMDIxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFhOWNhYTFkNTg2Yzc3MGIwYWI0MDhhY2EyZjlhM2U0NTUzYTQyNWYzMGNhNTNmZTRhNTBlZjBiNWVjNzJhYjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.16zlnj-JbyjdiqQOfo4TaNod5PF2hyetQEc6T_Zs1Bk) After downloading the print.sh file, I noticed such a symbol. Is this expected behavior? Initially, I thought it was because I used the / for searching, which caused this, but after exiting Yazi and searching for other files, I couldn't replicate this behavior

Do you have Nerd Fonts installed?

**Comment by evpeople** [comment]
Yes,
This is the result of my fc-list.
```
/Users/evpeople/Library/Fonts/SymbolsNerdFont-Regular.ttf: Symbols Nerd Font:style=Regular
/Users/evpeople/Library/Fonts/SymbolsNerdFontMono-Regular.ttf: Symbols Nerd Font Mono:style=Regular
/Users/evpeople/Library/Fonts/Go Mono Nerd Font Complete.ttf: GoMono Nerd Font:style=Book
/Users/evpeople/Library/Fonts/Go Mono Nerd Font Complete Mono.ttf: GoMono Nerd Font Mono:style=Book
```

**Comment by sxyazi** [comment]
What's your Nerd font version? Can it render properly, if you paste this line to your terminal?

https://github.com/sxyazi/yazi/blob/3f27532c576365bf5da4a82ee89c48a677ec498f/yazi-config/preset/theme.toml#L205

**Comment by sxyazi** [comment]
> > Could this be due to extended attributes in macOS? I noticed that the output of ls -al now includes an @ symbol.
> 
> I've also thought about it but they all have macOS attrs: <img alt="image" width="290" src="https://private-user-images.githubusercontent.com/46736350/305935919-a28cbf67-faeb-484a-adde-f56c644ffdf4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgzNDQ4OTAsIm5iZiI6MTcwODM0NDU5MCwicGF0aCI6Ii80NjczNjM1MC8zMDU5MzU5MTktYTI4Y2JmNjctZmFlYi00ODRhLWFkZGUtZjU2YzY0NGZmZGY0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE5VDEyMDk1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThjODMxMGRlOGUyZGExNTM5NzI0YTU1N2MwNGI4MmUxYWUyY2QwNDYzM2ZmZDg5MmI1ZDJhZmQ1ZWQ2ZDM1ZDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.m2gP492ATJLLc6rhGlhu0TIohqcyZeurhs3k6FpiyXQ"> <img alt="image" width="541" src="https://private-user-images.githubusercontent.com/46736350/305936220-9f05a058-9e42-41fc-970b-56f5c17db2eb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgzNDQ4OTAsIm5iZiI6MTcwODM0NDU5MCwicGF0aCI6Ii80NjczNjM1MC8zMDU5MzYyMjAtOWYwNWEwNTgtOWU0Mi00MWZjLTk3MGItNTZmNWMxN2RiMmViLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE5VDEyMDk1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgxOWE1NWQxODhlYmU3Y2FlYjdiMWViNjAwNGRmNTMxYmNkNTE0NTg2ZWEwZTRjMmUwYTVhZDFlN2QwZjA0ZTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.l2ttRMinyWr010kJUJr3JfMYYx7XnnDcCpSJe4StUMw"> (I've changed the attr of `print.sh`, but the issue still exist.)

Unfortunately, I can't reproduce it by using your file, could you try building Yazi from the main branch to see if the problem still? https://yazi-rs.github.io/docs/installation#build-from-source

**Comment by Saafo** [comment]
> Unfortunately, I can't reproduce it by using your file, could you try building Yazi from the main branch to see if the problem still? https://yazi-rs.github.io/docs/installation#build-from-source

Ok, I'll build it tomorrow. If this issue still exist, can I add some log at somewhere to find out how it happens?

**Comment by evpeople** [comment]
> *.sh" , text = "", fg = "#4d5a5e" },

Thank you, I found out it was because I hadn't configured the nerd font for Warp that I am using.

**Comment by evpeople** [comment]
> What's your Nerd font version? Can it render properly, if you paste this line to your terminal?
> 
> https://github.com/sxyazi/yazi/blob/3f27532c576365bf5da4a82ee89c48a677ec498f/yazi-config/preset/theme.toml#L205
> 
> It looks like this on my side:
> 
> ![CleanShot 2024-02-19 at 20 17 45](https://private-user-images.githubusercontent.com/17523360/305945709-046a8af8-ddbf-4984-9ba2-3c42edfa30b5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgzNDk5ODcsIm5iZiI6MTcwODM0OTY4NywicGF0aCI6Ii8xNzUyMzM2MC8zMDU5NDU3MDktMDQ2YThhZjgtZGRiZi00OTg0LTliYTItM2M0MmVkZmEzMGI1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE5VDEzMzQ0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJmMWI4MWYxNDNlZGFlZTdjZjExZjIxMDQyNjA4MjZkM2RjZmFhZWQyYmU2MWNkZThkYWJjMzg3NGQ4OWVmN2UmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.49xaFZ6YPnBKCKt-z6gkVf5n4pNLszFKf9UZmkBS-M8)

This is amazing, it's my first time discovering that Yazi has so many cute icons.

**Comment by Saafo** [comment]
> Unfortunately, I can't reproduce it by using your file, could you try building Yazi from the main branch to see if the problem still? https://yazi-rs.github.io/docs/installation#build-from-source

I've build from the latest main branch, and the issue still exist. Where can I add log or do something to find out why this happens?

**Comment by Saafo** [comment]
@sxyazi I thought there should exist an out-of-order issue:

I add a log at here:

https://github.com/sxyazi/yazi/blob/8efafd6c9104684920c819aa3fc762d9d61689e1/yazi-core/src/manager/commands/open.rs#L96

<img width="937" alt="image" src="https://github.com/sxyazi/yazi/assets/46736350/4a72f170-4d9a-4c03-aa1e-04bb13ff8881">

and here is the result:

| file | mime |
| --- | --- |
| print.sh | video/quicktime |
| print.swift | application/octet-stream |
| print2.sh | text/plain |
| mov.mp4 | text/x-script.python |

And here is the file list in this folder:
<img width="358" alt="image" src="https://github.com/sxyazi/yazi/assets/46736350/83191509-d3d2-41ee-99b4-0e65fd75de7a">

**Comment by sxyazi** [comment]
Please add logs next to this line and then see what happens:

https://github.com/sxyazi/yazi/blob/8efafd6c9104684920c819aa3fc762d9d61689e1/yazi-core/src/manager/commands/update_mimetype.rs#L54

**Comment by Saafo** [comment]
@sxyazi It's already incorrect. (some file name had been renamed because of privacy)

```yml
{
  /Users/saafo/Downloads/print.swift: "application/octet-stream",
  /Users/saafo/Downloads/mock_output.py: "image/avif",
  /Users/saafo/Downloads/usercard-detail.js: "text/plain",
  /Users/saafo/Downloads/xcb.swift: "text/plain",
  /Users/saafo/Downloads/print2.sh: "text/plain",
  /Users/saafo/Downloads/add_prefix.py: "text/x-script.python",
  /Users/saafo/Downloads/IMG_7254.AVIF: "video/quicktime",
  /Users/saafo/Downloads/mock_output.sh: "image/png",
  /Users/saafo/Downloads/mov.mp4: "text/x-script.python",
  /Users/saafo/Downloads/box4.stl: "application/octet-stream",
  /Users/saafo/Downloads/test.log: "text/plain",
  /Users/saafo/Downloads/bugheic.HEIC: "image/heic",
  /Users/saafo/Downloads/mess_thumb.png: "image/heic",
  /Users/saafo/Downloads/demo_thumb.png: "image/png",
  /Users/saafo/Downloads/somepdf.pdf: "text/x-shellscript",
  /Users/saafo/Downloads/Simulator Screen Recording.mp4: "video/mp4",
  /Users/saafo/Downloads/docckit: "application/x-mach-binary",
  /Users/saafo/Downloads/someapp.dmg: "text/x-shellscript",
  /Users/saafo/Downloads/xcb.sh: "text/plain",
  /Users/saafo/Downloads/print.sh: "video/quicktime",
}
```

**Comment by sxyazi** [comment]
Then apply this patch:

```diff
--- /yazi-plugin/preset/plugins/mime.lua
+++ /yazi-plugin/preset/plugins/mime.lua
@@ -3,6 +3,7 @@ local M = {}
 function M:preload()
 	local urls = {}
 	for _, file in ipairs(self.files) do
+		ya.err("file.url: " .. tostring(file.url))
 		urls[#urls + 1] = tostring(file.url)
 	end
 
@@ -33,6 +34,7 @@ function M:preload()
 			break
 		end
 
+		ya.err("next: " .. next)
 		next = next:gsub("[\r\n]+$", "")
 		if ya.mime_valid(next) then
 			j, mimes[urls[i]] = j + 1, next
```

And paste your `~/.local/state/yazi/yazi.log` content here

**Comment by Saafo** [comment]
@sxyazi 
<details>
<summary>log</summary>
  [2m2024-02-20T11:05:18.952905Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mfile.url: /Users/saafo/Downloads/SimulatorScreen Recording.mp4[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.952996Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mfile.url: /Users/saafo/Downloads/add_prefix.py[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.953006Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mfile.url: /Users/saafo/Downloads/box4.stl[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.953014Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mfile.url: /Users/saafo/Downloads/bugheic.HEIC[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.953022Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mfile.url: /Users/saafo/Downloads/demo_thumb.png[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.953030Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mfile.url: /Users/saafo/Downloads/docckit[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.953037Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mfile.url: /Users/saafo/Downloads/fmp4.mp4[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.953044Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mfile.url: /Users/saafo/Downloads/goodheic.HEIC[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.953051Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mfile.url: /Users/saafo/Downloads/IMG_7254.AVIF[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.953059Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mfile.url: /Users/saafo/Downloads/mess_thumb.png[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.953066Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mfile.url: /Users/saafo/Downloads/mock_output.py[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.953073Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mfile.url: /Users/saafo/Downloads/mock_output.sh[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.953081Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mfile.url: /Users/saafo/Downloads/mov.mp4[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.953088Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mfile.url: /Users/saafo/Downloads/someapp.dmg[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.953118Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mfile.url: /Users/saafo/Downloads/print.sh[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.953126Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mfile.url: /Users/saafo/Downloads/print.swift[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.953134Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mfile.url: /Users/saafo/Downloads/print2.sh[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.953144Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mfile.url: /Users/saafo/Downloads/test.log[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.953152Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mfile.url: /Users/saafo/Downloads/usercard-detail.js[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.953159Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mfile.url: /Users/saafo/Downloads/xcb.sh[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.953166Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mfile.url: /Users/saafo/Downloads/xcb.swift[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.953174Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mfile.url: /Users/saafo/Downloads/somepdf.pdf[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.986777Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mnext: video/mp4
[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.986851Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mnext: text/x-script.python
[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.986870Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mnext: application/octet-stream
[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.986886Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mnext: image/heic
[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.986902Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mnext: image/png
[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.986914Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mnext: application/x-mach-binary
[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.986926Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mnext: /Users/saafo/Downloads/docckit (for architecture x86_64):	application/x-mach-binary
[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.986939Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mnext: /Users/saafo/Downloads/docckit (for architecture arm64):	application/x-mach-binary
[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.986951Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mnext: video/quicktime
[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.986962Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mnext: image/heic
[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.986977Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mnext: image/avif
[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.986989Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mnext: image/png
[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.987003Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mnext: text/x-script.python
[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.987014Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mnext: text/x-shellscript
[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.987026Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mnext: video/quicktime
[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.987036Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mnext: application/octet-stream
[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.987047Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mnext: text/plain
[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.987056Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mnext: text/plain
[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.987094Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mnext: text/plain
[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.987106Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mnext: text/plain
[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.987115Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mnext: text/plain
[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.987125Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mnext: text/x-shellscript
[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.987423Z[0m [31mERROR[0m [1;31myazi_scheduler::preload::preload[0m[31m: [31mPreload task `mime` returned 2[0m
    [2;3mat[0m yazi-scheduler/src/preload/preload.rs:42

  [2m2024-02-20T11:05:18.987453Z[0m [32m INFO[0m [1;32myazi_core::manager::commands::update_mimetype[0m[32m: [32mmimes: ${/Users/saafo/Downloads/print2.sh: "text/plain", /Users/saafo/Downloads/xcb.sh: "text/plain", /Users/saafo/Downloads/print.swift: "application/octet-stream", /Users/saafo/Downloads/bugheic.HEIC: "image/heic", /Users/saafo/Downloads/docckit: "application/x-mach-binary", /Users/saafo/Downloads/mock_output.sh: "image/png", /Users/saafo/Downloads/someapp.dmg: "text/x-shellscript", /Users/saafo/Downloads/add_prefix.py: "text/x-script.python", /Users/saafo/Downloads/mock_output.py: "image/avif", /Users/saafo/Downloads/SimulatorScreen Recording.mp4: "video/mp4", /Users/saafo/Downloads/mov.mp4: "text/x-script.python", /Users/saafo/Downloads/test.log: "text/plain", /Users/saafo/Downloads/somepdf.pdf: "text/x-shellscript", /Users/saafo/Downloads/usercard-detail.js: "text/plain", /Users/saafo/Downloads/print.sh: "video/quicktime", /Users/saafo/Downloads/demo_thumb.png: "image/png", /Users/saafo/Downloads/mess_thumb.png: "image/heic", /Users/saafo/Downloads/xcb.swift: "text/plain", /Users/saafo/Downloads/IMG_7254.AVIF: "video/quicktime", /Users/saafo/Downloads/box4.stl: "application/octet-stream"}[0m
    [2;3mat[0m yazi-core/src/manager/commands/update_mimetype.rs:55

  [2m2024-02-20T11:05:18.990559Z[0m [31mERROR[0m [1;31myazi_scheduler::preload::preload[0m[31m: [31mPreload task `image` returned 2[0m
    [2;3mat[0m yazi-scheduler/src/preload/preload.rs:42

  [2m2024-02-20T11:05:18.990594Z[0m [31mERROR[0m [1;31myazi_scheduler::preload::preload[0m[31m: [31mPreload task `image` returned 2[0m
    [2;3mat[0m yazi-scheduler/src/preload/preload.rs:42

  [2m2024-02-20T11:05:18.990722Z[0m [31mERROR[0m [1;31myazi_scheduler::preload::preload[0m[31m: [31mPreload task `image` returned 2[0m
    [2;3mat[0m yazi-scheduler/src/preload/preload.rs:42

  [2m2024-02-20T11:05:18.993358Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mspawn `ffmpegthumbnailer` command returns 2[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.993478Z[0m [31mERROR[0m [1;31myazi_scheduler::preload::preload[0m[31m: [31mPreload task `video` returned 0[0m
    [2;3mat[0m yazi-scheduler/src/preload/preload.rs:42

  [2m2024-02-20T11:05:18.995112Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mspawn `ffmpegthumbnailer` command returns 2[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.995315Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mspawn `ffmpegthumbnailer` command returns 2[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.995420Z[0m [31mERROR[0m [1;31myazi_scheduler::preload::preload[0m[31m: [31mPreload task `video` returned 0[0m
    [2;3mat[0m yazi-scheduler/src/preload/preload.rs:42

  [2m2024-02-20T11:05:18.996002Z[0m [31mERROR[0m [1;31myazi_plugin::utils::log[0m[31m: [31mspawn `ffmpegthumbnailer` command returns 2[0m
    [2;3mat[0m yazi-plugin/src/utils/log.rs:10

  [2m2024-02-20T11:05:18.996110Z[0m [31mERROR[0m [1;31myazi_scheduler::preload::preload[0m[31m: [31mPreload task `video` returned 0[0m
    [2;3mat[0m yazi-scheduler/src/preload/preload.rs:42
</details>

**Comment by sxyazi** [comment]
I ground it a bit, starting from the `docckit` file, the file generated inconsistent output lines:

```
/Users/saafo/Downloads/SimulatorScreen Recording.mp4  :  video/mp4
/Users/saafo/Downloads/add_prefix.py                  :  text/x-script.python
/Users/saafo/Downloads/box4.stl                       :  application/octet-stream
/Users/saafo/Downloads/bugheic.HEIC                   :  image/heic
/Users/saafo/Downloads/demo_thumb.png                 :  image/png
/Users/saafo/Downloads/docckit                        :  application/x-mach-binary
/Users/saafo/Downloads/fmp4.mp4                       :  /Users/saafo/Downloads/docckit (for architecture x86_64): application/x-mach-binary
/Users/saafo/Downloads/goodheic.HEIC                  :  /Users/saafo/Downloads/docckit (for architecture arm64): application/x-mach-binary
/Users/saafo/Downloads/IMG_7254.AVIF                  :  video/quicktime
/Users/saafo/Downloads/mess_thumb.png                 :  image/heic
/Users/saafo/Downloads/mock_output.py                 :  image/avif
/Users/saafo/Downloads/mock_output.sh                 :  image/png
/Users/saafo/Downloads/mov.mp4                        :  text/x-script.python
/Users/saafo/Downloads/someapp.dmg                    :  text/x-shellscript
/Users/saafo/Downloads/print.sh                       :  video/quicktime
/Users/saafo/Downloads/print.swift                    :  application/octet-stream
/Users/saafo/Downloads/print2.sh                      :  text/plain
/Users/saafo/Downloads/test.log                       :  text/plain
/Users/saafo/Downloads/usercard-detail.js             :  text/plain
/Users/saafo/Downloads/xcb.sh                         :  text/plain
/Users/saafo/Downloads/xcb.swift                      :  text/plain
/Users/saafo/Downloads/somepdf.pdf                    :  text/x-shellscript
```

What kind of file is it? Is it possible to send it to me for testing (if it is not privacy-related)? 

Also please do `file -bL --mime-type /Users/saafo/Downloads/demo_thumb.png /Users/saafo/Downloads/docckit /Users/saafo/Downloads/fmp4.mp4` then paste the output here.

**Comment by Saafo** [comment]
@sxyazi Sorry this file is privacy-related, I cannot share it. But it's a unix executable binary file compiled from swift(a command line program).

<img width="993" alt="image" src="https://github.com/sxyazi/yazi/assets/46736350/6bfffb9c-5d9d-42f2-a1c8-6e4142b6358b">

It seems that the docckit file is a fat/universal/multi-arch binary, so the mime result output is multi line!

**Comment by sxyazi** [comment]
Hi, I made a PR trying to fix it, https://github.com/sxyazi/yazi/pull/707, let me know if it works for you!

**Comment by Saafo** [comment]
@sxyazi The unix executable file with multi arches support: [xpm.zip](https://github.com/sxyazi/yazi/files/14353506/xpm.zip)

**Comment by Saafo** [comment]
> Hi, I made a PR trying to fix it, #707, let me know if it works for you!

Oh it works! Thanks for the fix!

**Comment by sxyazi** [comment]
Thanks for testing! Let's merge it then
